### PR TITLE
Feat/heatmap home view

### PR DESCRIPTION
**Pull request by svandecappelle**
1. Description
Add a screen of workflow runs status. Added on home page into heatmap tab.
The aim is to have a most usable screen than the timeline.
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report it#7808.0 ✘  
* w-it
  * w-it ✘

File: ui/src/app/views/home/heatmap/home.heatmap.component.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            if (this.timelineSub) {
+                this.timelineSub.unsubscribe();
+            }
+            if (f) {
+
+                this.timelineSub = this._timelineStore.alltimeline().subscribe(es => {
+                    if (!es) {
+                        return;
+                    }
+                    this.loading = false;
```
Comment (bnjjj) [inline ui/src/app/views/home/heatmap/home.heatmap.component.ts]: put it in 
`.pipe(finalize(() => this.loading = false)))`

File: ui/src/app/views/home/timeline/home.timeline.html [hunk 2]
```diff
@@ -1,50 +1,40 @@
...
-                                {{e.timestamp | amTimeAgo}}
+
+<ng-container *ngIf="!loading && events.length > 0">
+    <div class="ui relaxed divided list"
+            infiniteScroll
+            [infiniteScrollDistance]="2"
+            [infiniteScrollThrottle]="50"
+            [scrollWindow]="false"
+            (scrolled)="onScroll()">
+        <div class="item" *ngFor="let e of events" (mouseenter) ="e.bell = true"  (mouseleave) ="e.bell = false">
```
Comment (bnjjj) [inline ui/src/app/views/home/timeline/home.timeline.html]:
```suggestion
         <div class="item" *ngFor="let e of events" (mouseenter)="e.bell = true"  (mouseleave)="e.bell = false">
```

File: ui/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -408,6 +408,13 @@
   "token_group_name": "Groupe",
   "token_generated": "Voici votre token généré",
 
+  "heatmap": "Carte des évènements",
+  "heatmap_empty": "Aucun évènement déclanché",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]:
```suggestion
   "heatmap_empty": "Aucun évènement déclenché",
```

File: ui/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -408,6 +408,13 @@
...
+  "heatmap_filter_mute": "Masquer les évènements de ce projet",
+  "heatmap_status": "Statut",
+  "heatmap_timestamp": "Date d'évenement",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]:
```suggestion
   "heatmap_timestamp": "Date d'évènement",
```

File: ui/src/app/views/home/heatmap/home.heatmap.component.ts [hunk 5]
```diff
@@ -35,72 +38,83 @@ export class HomeHeatmapComponent implements OnInit {
...
-                        if (!this.workflows[event.project_key]) {
-                            this.workflows[event.project_key] = new Array<string>();
-                        }
-                        if (this.workflows[event.project_key].lastIndexOf(event.workflow_name) === -1) {
-                            this.workflows[event.project_key].push(event.workflow_name);
-                        }
-                    });
+                            if (eventsWorkflow.length > 0 && event.timestamp > eventsWorkflow[0].timestamp) {
+                                eventsWorkflow = eventsWorkflow.filter((pendingEvent: Event) => {
+                                    if (event.workflow_name === pendingEvent.workflow_name
```
Comment (richardlt) [inline ui/src/app/views/home/heatmap/home.heatmap.component.ts]: return directly the boolean expression

**Comment by yesnault** [comment]
@svandecappelle tags added on master

**Pull request merged**